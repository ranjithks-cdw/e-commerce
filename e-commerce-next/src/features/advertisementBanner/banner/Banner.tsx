import Link from "next/link";
import Image from "next/image";

import { adBannerType } from "../types/adBannerTypes";

const Banner = (props: adBannerType) => {
    const {adImageUrl, adName, adRedirectUrl} = props;

    return ( 
        <Link href={adRedirectUrl}>
            <div className="advertisement-banner">
                <Image src={adImageUrl} alt={adName} width={0} height={0} sizes="100vw" priority />
            </div>
        </Link>
     );
}
 
export default Banner;