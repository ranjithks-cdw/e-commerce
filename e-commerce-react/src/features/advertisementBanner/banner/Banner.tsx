import { adBannerType } from "../types/adBannerTypes";

const Banner = (props: adBannerType) => {
    const {adImageUrl, adName, adRedirectUrl} = props;

    return ( 
        <a href={adRedirectUrl}>
            <div className="advertisement-banner">
                <img src={adImageUrl} alt={adName} />
            </div>
        </a>
     );
}
 
export default Banner;