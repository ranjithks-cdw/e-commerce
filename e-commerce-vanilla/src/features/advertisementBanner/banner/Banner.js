import { createElement } from "../../../utils/htmlUtils.js";

const Banner = {
    create: (props) => createDOM(props),
};

const createDOM = (props) => {
    const {adImageUrl, adName, adRedirectUrl} = props;

    const banner = createElement('a', {
        href: adRedirectUrl
    });

    const adBanner = createElement('div', {
        className: 'advertisement-banner'
    });

    const image = createElement('img', {
        src: adImageUrl,
        alt: adName
    });

    adBanner.appendChild(image);
    banner.appendChild(adBanner);
    return banner;
};

export default Banner;