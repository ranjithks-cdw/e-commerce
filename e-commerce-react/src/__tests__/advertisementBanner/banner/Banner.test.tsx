import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import Banner from "../../../features/advertisementBanner/components/banner/Banner";

const adData = {
    adName: "Title 1",
    adImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
    adRedirectUrl: "/products"
};

describe('Banner', () => {
    it("renders component", async () => {
        render(
            <MemoryRouter>
                <Banner {...adData} />
            </MemoryRouter>
        );
        const image = await waitFor(() => screen.findByAltText(adData.adName), {timeout: 2000});
        expect(image).not.toBeNull();
        expect(image).toHaveAttribute('src', adData.adImageUrl);
    })

    it("matches snapshot", () => {
        const container = renderer.create(
            <MemoryRouter>
                <Banner {...adData} />
            </MemoryRouter>
        ).toJSON;
        expect(container).toMatchSnapshot();
    })
})