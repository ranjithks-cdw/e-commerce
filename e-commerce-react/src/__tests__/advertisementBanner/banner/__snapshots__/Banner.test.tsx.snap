// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches snapshot 1`] = `[Function]`;
